### PR TITLE
Update documentation with purpose and scope

### PR DESCRIPTION
**Pull request by piercebb**
## Background
The `fake-eggs` module is used to generate fake data, but [the scope of the module has been undefined](https://github.com/goodeggs/fake-eggs/issues/42). There have been discussions on whether to "bless" `fake-eggs` as our go-to library for generating fake data or to use an external non-good eggs specific library (e.g. [Faker](https://faker.readthedocs.io/en/master/) and [Chance.js](https://chancejs.com/index.html)).

See discussions [here](https://github.com/goodeggs/fake-eggs/issues/41), and [here](https://github.com/goodeggs/standards-and-best-practices/issues/272). 

After [deliberation within the Standards and Best Practices Guild](https://goodeggs.atlassian.net/wiki/spaces/TECH/pages/891519258/2020-10-28+Standards+and+Best+Practices+Guild+Meeting+notes) we have decided to move forward with scoping `fake-eggs` as the go-to random factory data generator for a few reasons: 

1. Using `fake-eggs` as the primary way to generate fake data is already practiced by the engineering team
2. Simplifies test data generation
3. "Blessing" this test module enables the engineering team to manage/control our test data library
4. Supports `fake-eggs` as a custom wrapper for integrating other test libraries in the future

## Changes
- Update documentation to define the `fake-eggs` module as the go-to for all random factory data

## TODO
- Update [Standards and Best Practices](https://github.com/goodeggs/standards-and-best-practices) to reflect these changes

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ Providing you with all sorts of Good Eggs-style fake data!
 
 All exported functions are stateless (meaning you don't have to worry about binding to `this`).
 
+## Purpose and Scope
+
+Use `fake-eggs` to generate all random factory data except for json schemas where [unionized](https://github.com/goodeggs/unionized) should be employed.
```
Comment (serhalp) [inline readme.md:9]: and mongoose schemas?

File: readme.md [hunk 2]
```diff
@@ -4,6 +4,10 @@ Providing you with all sorts of Good Eggs-style fake data!
 
 All exported functions are stateless (meaning you don't have to worry about binding to `this`).
 
+## Purpose and Scope
```
Comment (serhalp) [inline readme.md:7]: Maybe it's implied in this paragraph as is, but I wonder if we could be more explicit about the intended action to take when your use case can't be adequately covered by fake-eggs? Something like:
> Use fake-eggs to generate all [...]
> 
> If you encounter a substantial use case where fake-eggs is unable to expressively generate random factory data, please consider opening a PR enhancing it or opening an Issue to track it.

File: readme.md [hunk 3]
```diff
@@ -4,6 +4,10 @@ Providing you with all sorts of Good Eggs-style fake data!
...
+
+Use `fake-eggs` to generate all random factory data except for json schemas where [unionized](https://github.com/goodeggs/unionized) should be employed.
+
```
Comment (serhalp) [inline readme.md:10]: We had discussed whether to bless fake-eggs for non-test use cases where random realistic data is necessary (e.g. sanitizing a dev database, populating mock data for local development, etc.). Should we add a few words here calling out that its intended scope isn't limited to tests? It already doesn't technically _say_ it's limited to tests, but I can see it being assumed to be implied by some readers.

File: readme.md [hunk 4]
```diff
@@ -4,6 +4,10 @@ Providing you with all sorts of Good Eggs-style fake data!
 
 All exported functions are stateless (meaning you don't have to worry about binding to `this`).
 
+## Purpose and Scope
+
```
Comment (serhalp) [inline readme.md:8]: (Sorry for putting comments on random lines – I want to be able to have a proper thread for each comment...)

To me, a fundamental purpose of fake-eggs is to generated **realistic** or **plausible** fake data. This is as opposed to using a generic fake data library (directly), which would generate more general purpose categories of fake data. Could we document this as well?

**Review by serhalp — COMMENTED** [review]
Thanks for taking this on! See inline prompts. (Sorry for the delayed review!)

File: readme.md (hunk 2 above)
Comment (piercebb) [reply readme.md:7]: Good idea - I support a call-to-action of sorts!

File: readme.md (hunk 3 above)
Comment (piercebb) [reply readme.md:14]: What do you think of this line? Encouraging enough for the use cases beyond test data? 

> This module can be used for test data, local development, and beyond!

**Review by serhalp — APPROVED** [review]
🚀

**Review by mattpetrie — COMMENTED** [review]
Documenting some of my thoughts that I expressed at the S&BP Guild meeting today here:

I think this change greatly improves clarifying the _purpose_ of fake-eggs. But, I still think we need to be more specific about the _scope_.

Specifically, I'm referring to the concerns raised in https://github.com/goodeggs/standards-and-best-practices/issues/272. Is fake-eggs an option that you _can_  use when you require fake data (in other words, it would be acceptable for someone to choose to use faker.js instead if they wanted to), or is it something you _must_ use? I think we need to unambiguously state an answer to this.

When we've discussed this previously, it sounded like most folks were on board with making fake-eggs our default fake data library (prefer fake-eggs over other fake data libraries whenever possible). 

However, if that's our stance, we need to clearly state what to do when fake-eggs doesn't cover your need, and we need to recommend a solution that can be employed immediately. If I am in the middle of writing a test, and come here because I'm having trouble using fake-eggs for the data I need, I need to know what do right away so I can get on with my work. Do I use a different library (and if so, is there an endorsed choice of what to fall back on)? Am I expected to submit a PR here and wait for it to be merged before completing my test (I'd hope not.)?  This is a situation that is virtually guaranteed to occur unless we greatly expand fake-eggs. Given that even if we wanted to expand fake-eggs, it won't occur for some time, I think we need to provide an answer here now, e.g. "Prefer using fake-eggs whenever possible. When fake-eggs doesn't provide the type of data you need, use chance.js".

**Pull request merged**